#==============================================================================
# Patch to fix reference to google-glog's IsGoogleLoggingInitialized().
# In versions prior to 0.5.0, that function was in namespace
# 'google::glog_internal_namespace_'. Then the function was hidden in 0.5.0,
# breaking dependents which rely on it.
#
# Meanwhile, google-glog has an upstream fix, which exposes that API via
# public namespace 'google'. (And MacPorts includes that in our 'google-glog'
# port.)
#
# As for PyTorch, it still relies on the internal namespace, which doesn't
# work with google-glob 0.5.0. So, patch to utilize public API, until PyTorch
# upstream provides a formal fix.
#
# -CNielsen 2022-02-16
#==============================================================================
*** a/c10/util/Logging.cpp.orig	2022-02-16 11:33:05.000000000 -0500
--- b/c10/util/Logging.cpp	2022-02-16 11:34:08.000000000 -0500
***************
*** 168,194 ****
  C10_DEFINE_int(
      caffe2_log_level,
      google::GLOG_WARNING,
      "The minimum log level that caffe2 will output.");
  
- // Google glog's api does not have an external function that allows one to check
- // if glog is initialized or not. It does have an internal function - so we are
- // declaring it here. This is a hack but has been used by a bunch of others too
- // (e.g. Torch).
- namespace google {
- namespace glog_internal_namespace_ {
- bool IsGoogleLoggingInitialized();
- } // namespace glog_internal_namespace_
- } // namespace google
- 
  namespace c10 {
  bool InitCaffeLogging(int* argc, char** argv) {
    if (*argc == 0)
      return true;
  #if !defined(_MSC_VER)
    // This trick can only be used on UNIX platforms
!   if (!::google::glog_internal_namespace_::IsGoogleLoggingInitialized())
  #endif
    {
      ::google::InitGoogleLogging(argv[0]);
  #if !defined(_MSC_VER)
      // This is never defined on Windows
--- 168,184 ----
  C10_DEFINE_int(
      caffe2_log_level,
      google::GLOG_WARNING,
      "The minimum log level that caffe2 will output.");
  
  namespace c10 {
  bool InitCaffeLogging(int* argc, char** argv) {
    if (*argc == 0)
      return true;
  #if !defined(_MSC_VER)
    // This trick can only be used on UNIX platforms
!   if (!::google::IsGoogleLoggingInitialized())
  #endif
    {
      ::google::InitGoogleLogging(argv[0]);
  #if !defined(_MSC_VER)
      // This is never defined on Windows
